src/chaat/db/friend_request.clj

```diff
@@ -0,0 +1,60 @@
+(ns chaat.db.friend-request
+  (:require
+   [next.jdbc :as jdbc]
+   [next.jdbc.sql :as sql]
+   [chaat.handler.errors :refer [error-table]]
+   [chaat.db.user :as user]
+   [java-time.api :as jt]))
+
+;; Will be more logical to have the user-exists? and exists? check in the model layer.
```

Comment:
There's a tradeoff here.

- Putting it in the model layer means you're able to visualize the flow better at the correct layer. Putting it in the db layer lets you use db features to do this better.
- If you're doing the check in the model layer, you'll have to use a transaction. Otherwise, you'll have to check again in the db layer anyway (or implicitly, it will fail the uniqueness check which you should handle)
- Just relying on the uniqueness check, will allow you to skip any explicit checks entirely. The downside to this is that it isn't clear from just the code that there is a uniqueness guarantee. However, this may be okay since any complex enough system is relying on the db for a variety of guarantees. It depends on how much you want to tie your app to DB features. The more you rely on them, the more valuable the dependency (but also the more dependent you are and moving away will be harder).